lib/project-parser.ts

```diff
@@ -29,26 +32,26 @@ export class ProjectParser {
   events = new EventEmitter();
   private watchers: FSWatcher[] = [];
   // Constructor can not be async. So constructor is private and use factory to create
-  public static async create(workspaces: string[], fileIgnoreRegex: string, settingsGetter: SettingsGetter, disableWatching = false) {
+  public static async create(workspaces: URL[], fileIgnoreRegex: string, settingsGetter: SettingsGetter, disableWatching = false) {
     const projectParser = new ProjectParser(workspaces, fileIgnoreRegex, settingsGetter);
     await projectParser.init(disableWatching);
     return projectParser;
   }
-  private constructor(public workspaces: string[], public fileIgnoreRegex: string, public settingsGetter: SettingsGetter) { }
-  public addFolders(paths: string[]) {
-    for (const path of paths) {
-      const watcher = watch(path.replace(sep, '/') + '/**/*.vhd?(l)', { ignoreInitial: true });
+  private constructor(public workspaces: URL[], public fileIgnoreRegex: string, public settingsGetter: SettingsGetter) { }
+  public addFolders(urls: URL[]) {
+    for (const url of urls) {
+      const watcher = watch(fileURLToPath(url).replace(sep, '/') + '/**/*.vhd?(l)', { ignoreInitial: true });
```

Comment:
How does this `watch` behave on win systems?
At least the `.replace(sep, '/')` doesn't make sense, does it?